### PR TITLE
feat: in-memory watermark store for better testing

### PR DESCRIPTION
**Pull request by chromevoid**
watermark unit tests with in mem store

### CODE REVIEW

File: pkg/watermark/store/inmem/kv_watch.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	return updates
+}
+
+// GetKVName returns the KV store (bucket) name.
+func (k *inMemWatch) GetKVName() string {
+	return k.bucketName
+}
+
+// Close closes the connection.
+func (k *inMemWatch) Close() {
```
Comment (vigith) [inline pkg/watermark/store/inmem/kv_watch.go:99]: Can you inspect how `Close` is invoked? We do have `Close` in the interface. Can you make sure we are using `Close`? Without correctly closing, we might run into connection leaks and memory leaks associated with it.

Comment (chromevoid) [reply pkg/watermark/store/inmem/kv_watch.go:99]: 👌

Comment (chromevoid) [reply pkg/watermark/store/inmem/kv_watch.go:99]: opened a new issue https://github.com/numaproj/numaflow/issues/224

**Review by vigith — APPROVED** [review]

File: pkg/watermark/store/inmem/kv_watch.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
+package inmem
+
+import (
+	"context"
+	"math/rand"
```
Comment (whynowy) [inline pkg/watermark/store/inmem/kv_watch.go:5]: No `math/rand`

File: pkg/watermark/store/inmem/kv_watch.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		kvHistory:    []store.WatermarkKVEntry{},
+		updatesChMap: make(map[string]chan store.WatermarkKVEntry),
+		log:          logging.FromContext(ctx).With("pipeline", pipelineName).With("bucketName", bucketName),
+	}
+	return k, nil
+}
+
+var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+func randSeq(n int) string {
```
Comment (whynowy) [inline pkg/watermark/store/inmem/kv_watch.go:42]: use https://github.com/numaproj/numaflow/blob/main/pkg/shared/util/string.go#L10

**Pull request merged**